src/main/java/seedu/address/model/time/TimeMask.java

```diff
@@ -70,17 +70,26 @@ public void mergeIsolatedEvents(IsolatedEventList isolatedEventList, LocalDate s
 
             if (endTime < 0) {
                 endTime = LAST_HOUR_INDEX;
+                endIndex = getUpdatedEndDay(endIndex);
             }
 
             if (startIndex != endIndex) {
```

Comment:
Just checking, would there be a bug if we start on Tuesday and we end a week later on Tuesday as well